Library/Homebrew/cask/cask_loader.rb

```diff
@@ -150,18 +146,10 @@ def self.can_load?(ref)
         super && !Tap.from_path(ref).nil?
       end
 
-      attr_reader :tap
-
       def initialize(path)
         @tap = Tap.from_path(path)
         super(path)
       end
-
-      private
-
-      def cask(*args, &block)
-        super(*args, tap: tap, &block)
-      end
```

Comment:
We can remove these parts because `@tap` is now defined in the `FromContentLoader` and passed into the cask constructor from there.

Easy to verify by doing the following. Note that the `FromDefaultTapPathLoader` inherits from the `FromTapPathLoader`.

```rb
/u/l/Homebrew (delay-loading-from-cask-source-api|✔) $ brew irb
==> Interactive Homebrew Shell
Example commands available with: `brew irb --examples`
irb(main):001:0> cask = Cask::CaskLoader::FromDefaultTapPathLoader.new("virtualbox").load(config: nil); nil
=> nil
irb(main):002:0> pp cask.tap; nil
#<Tap:0x00007f9ff2a13530
 @alias_reverse_table=nil,
 @alias_table=nil,
 @cask_dir=
  #<Pathname:/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks>,
 @full_name="Homebrew/homebrew-cask",
 @name="homebrew/cask",
 @path=#<Pathname:/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask>,
 @repo="cask",
 @user="Homebrew">
=> nil
```